domain/machine/state/reboot.go

```diff
@@ -0,0 +1,235 @@
+	unsetRebootFlagStmt, err := sqlair.Prepare(unsetRebootFlag, machineUUID{})
+	if err != nil {
+		return errors.Trace(err)
+	}
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		return tx.Query(ctx, unsetRebootFlagStmt, machineUUID{uuid}).Run()
+	})
+	return errors.Annotatef(err, "cancelling reboot of machine %q", uuid)
+}
+
+// IsMachineRebootRequired checks if the specified machine requires a reboot.
+//
+// It queries the "machine_requires_reboot" table for the machine UUID to determine if a reboot is required.
+// Returns a boolean value indicating if a reboot is required, and an error if any occur during the process.
+func (st *State) IsMachineRebootRequired(ctx context.Context, uuid string) (bool, error) {
+	db, err := st.DB()
+	if err != nil {
+		return false, errors.Trace(err)
+	}
+
+	var isRebootRequired bool
+	isRebootFlag := `SELECT machine_uuid as &machineUUID.uuid  FROM machine_requires_reboot WHERE machine_uuid = $machineUUID.uuid`
+	isRebootFlagStmt, err := sqlair.Prepare(isRebootFlag, machineUUID{})
+	if err != nil {
+		return false, errors.Trace(err)
+	}
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		var results machineUUID
+		err := tx.Query(ctx, isRebootFlagStmt, machineUUID{uuid}).Get(&results)
+		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
+			return errors.Trace(err)
+		}
+		isRebootRequired = !errors.Is(err, sqlair.ErrNoRows)
+		return nil
+	})
+
+	return isRebootRequired, errors.Annotatef(err, "requiring reboot of machine %q", uuid)
+}
+
+// GetParentMachine fetches the UUID of the parent machine based on the provided UUID.
+// It returns the UUID of the parent machine if found, otherwise returns an errors.NotFound.
+//
+// The function uses a database connection from the State struct to execute the query.
+// The `ctx` parameter is the context.Context used for the transaction.
+// The `uuid` parameter is the UUID of the machine to get the parent machine for.
+//
+// If the database query returns no rows, the function returns a errors.NotFound error
+// Any other errors that occur during the execution of the function are annotated with the UUID of the machine being fetched.
+func (st *State) GetParentMachine(ctx context.Context, uuid string) (string, error) {
+	db, err := st.DB()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	var parentMachine machineUUID
+	getParentQuery := `SELECT machine_parent.parent_uuid as &machineUUID.uuid  FROM machine_parent WHERE machine_uuid = $machineUUID.uuid`
+	getParentStmt, err := sqlair.Prepare(getParentQuery, machineUUID{})
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, getParentStmt, machineUUID{uuid}).Get(&parentMachine)
+		return errors.Trace(err)
+	})
+	if errors.Is(err, sqlair.ErrNoRows) {
+		err = errors.NotFoundf("parent machine not found")
```

Comment:
**issue (blocking):** We're moving away from the generic errors and instead, should use a sentinel error. See: 

https://github.com/juju/juju/blob/44c0db2057f43a5106a247adede929b2403f6bdc/domain/machine/errors/errors.go#L10-L18